server/src/main/java/io/druid/segment/realtime/appenderator/Appenderator.java

```diff
@@ -81,6 +81,14 @@ int add(SegmentIdentifier identifier, InputRow row, Supplier<Committer> committe
   List<SegmentIdentifier> getSegments();
 
   /**
+   * Indicates that the given segment exists.
+   *
+   * @param identifier segment to examine
+   * @return true if exists.
+   */
+  boolean containsSegment(SegmentIdentifier identifier);
```

Comment:
Right. I wanted to avoid copying segments every time, but removed now because it isn't used anymore.